hc-dna/zomes/perspective_diff_sync/src/pull.rs

```diff
@@ -155,6 +155,25 @@ pub fn pull<Retriever: PerspectiveDiffRetreiver>() -> SocialContextResult<Perspe
     }
 }
 
+pub fn fast_forward_signal<Retriever: PerspectiveDiffRetreiver>(revision: Hash) -> SocialContextResult<()> {
+    let diff = Retriever::get::<PerspectiveDiffEntryReference>(current_hash.clone())?;
+    let current_revision = current_revision()?;
+
+    if current_revision.is_some() {
+        if diff.parents.contains(&current_revision) {
```

Comment:
`.contains()` is not the correct condition, I believe. It should be `== Some(&current_revision)` because if the next entry is a merge with a side-branch, we need to pull as well.